server/controllers/auth.rb

```diff
@@ -58,7 +62,43 @@ class AuthController < ApplicationController
   end
   get '/me' do
     requires_login!
-    principal.to_json(include: :roles)
+    principal.to_json(include: [:roles,:avatar_url], except: [:id, :active, :password, :salt, :avatar])
+  end
+  post '/me' do
+    requires_login!
+    user_param_presence!
+    user = User[principal.id]
+
+    if params[:user][:passwordCurrent]
+      error_message = { :passwordCurrent => [ 'incorrect password' ] }
+      json_halt 400, error_message unless user.password_matches?(params[:user][:passwordCurrent])
```

Comment:
Yeah, while writing this I knew it was gross. What do you think of something like this, where we hook the Sequel validation logic?

```
   if params[:user][:passwordCurrent]                                                           
     user.update_password params[:user][:passwordCurrent], params[:user][:password]
     json_halt 400, user.errors unless user.valid?
```